### PR TITLE
WC Bump & refactor

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-####

## What changed (plus any additional context for devs)
- Prevent multiple initialization
- Remove duplicate listeners
- Bump SDK  to latest version

For more context: https://rainbowhaus.slack.com/archives/C02SU6XKFD0/p1724742752294839

## Screen recordings / screenshots

https://github.com/user-attachments/assets/5d491f1d-5d81-426a-9b6c-453ee2ae5d3e

## What to test

WC v2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -136,11 +136,11 @@
...
-    "@walletconnect/core": "2.11.2",
+    "@walletconnect/core": "2.15.1",
     "@walletconnect/legacy-utils": "2.0.0",
-    "@walletconnect/react-native-compat": "2.11.2",
-    "@walletconnect/types": "2.11.2",
-    "@walletconnect/utils": "2.11.2",
+    "@walletconnect/react-native-compat": "2.15.1",
+    "@walletconnect/types": "2.15.1",
+    "@walletconnect/utils": "2.15.1",
     "@walletconnect/web3wallet": "1.10.2",
```
Comment (ignaciosantise) [inline package.json:144]:
```suggestion
    "@walletconnect/web3wallet": "1.14.1",
```

File: src/walletConnect/index.tsx [hunk 2]
```diff
@@ -95,32 +96,46 @@ export function maybeGoBackAndClearHasPendingRedirect({ delay = 0 }: { delay?: n
...
-    icons: ['https://avatars2.githubusercontent.com/u/48327834?s=200&v=4'],
-    redirect: {
-      native: 'rainbow://wc',
-      universal: 'https://rnbwapp.com/wc',
-    },
-  },
-});
+let lastConnector: string | undefined = undefined;
+
+let walletConnectCore: TypeCore | undefined;
```
Comment (walmat) [inline src/walletConnect/index.tsx:101]: #nit use `ReturnType<typeof Core>` instead of importing the type from that internal path

**Review by walmat — APPROVED** [review]
hot. thanks for doing this. one smol nit but not blocking obv

File: src/walletConnect/index.tsx (hunk 2 above)
Comment (brunobar79) [reply src/walletConnect/index.tsx:101]: Tried that but the linter complains:

```
src/walletConnect/index.tsx:100:35 - error TS2344: Type 'typeof Core' does not satisfy the constraint '(...args: any) => any'.
  Type 'typeof Core' provides no match for the signature '(...args: any): any'.
  ```

Comment (brunobar79) [reply src/walletConnect/index.tsx]: nvm just found the right type!

**Pull request merged**